doc/how-to-source-packages.rst

```diff
@@ -0,0 +1,65 @@
+How to source packages
+======================
+
+Package Source
+--------------
+
+When we publish a package, its source is available from the package repository
+[#]_. The other way to access a package's source is from a source repository [#]_.
+
+Source Repositories
+-------------------
+
+Source repositories are a way to specify where to find the source code for a
+package, both for :ref:`package authors<pkg-authors>` and maintainers and for
+:ref:`package consumers<pkg-consumers>`.
+
+A relatively structured set of version control system (VCS) fields, that vary
+depending on the :ref:`VCS kind<vcs-kind>`, enables cabal commands and other
+tools to interpret and make effective use of this information.
+
+- ``source-repository`` says where to find the source for a **package description**
+- ``source-repository-package`` locates source for **project dependencies**
```

Comment:
Better "locates the source code to be added as a **project dependency**"

Reply:
I went with this suggestion, change the list to be a description list and added a warning about the confusing names.